power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <numeric>
+
+#include <iostream>
+#include "branch.hpp"
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+#include "../common/three_phase_tensor.hpp"
+#include "../common/matrix_utils.hpp"
+#include "line_utils.hpp"
+
+namespace power_grid_model {
+
+class AsymLine : public Branch {
+  public:
+    using InputType = AsymLineInput;
+    using UpdateType = BranchUpdate;
+    static constexpr char const* name = "asym_line";
+
+    explicit AsymLine(AsymLineInput const& asym_line_input, double system_frequency, double u1, double u2)
+        : Branch{asym_line_input}, i_n_{asym_line_input.i_n}, base_i_{base_power_3p / u1 / sqrt3} {
+        if (cabs(u1 - u2) > numerical_tolerance) {
+            throw ConflictVoltage{id(), from_node(), to_node(), u1, u2};
+        }
+
+        ComplexTensor<asymmetric_t> c_matrix = compute_c_matrix_from_input(asym_line_input);
+        ComplexTensor<asymmetric_t> z_series = compute_z_series_from_input(asym_line_input);
+
+        double const base_y = base_i_ / (u1 / sqrt3); 
+
+        y_series_abc_ = 1 / base_y * inv(z_series);
+        y_shunt_abc_ = 1 / base_y * (2 * pi * system_frequency * c_matrix * 1.0i);
+
+    }
+
+    // override getter
+    constexpr double base_i_from() const override { return base_i_; }
+    constexpr double base_i_to() const override { return base_i_; }
+    constexpr double loading(double /* max_s */, double max_i) const override { return max_i / i_n_; };
+    constexpr double phase_shift() const override { return 0.0; }
+    constexpr bool is_param_mutable() const override { return false; }
+
+  private:
+    double i_n_;
+    double base_i_;
+    ComplexTensor<asymmetric_t> y_series_abc_;
+    ComplexTensor<asymmetric_t> y_shunt_abc_;
```

Comment:
although you did it correctly in the constructor, safer to add default initialization
```suggestion
    double i_n_{};
    double base_i_{};
    ComplexTensor<asymmetric_t> y_series_abc_{};
    ComplexTensor<asymmetric_t> y_shunt_abc_{};
```

Reply:
Added it, could you please elaborate on why it is safer to add the default initialization here? Is that not done by default?

Reply:
nope, primitive types (i.e., `double`, `int`, `char`, ...) all do not have constructors. As a result, statements like `double a;` or `int b;` only reserve memory, without initializing it.

While it enables efficient low-level code, reading from those values are undefined behavior. That allows all sorts of weird things, ranging from reading values that should not exist anymore to hard crashes, and because the behavior is undefined, it may differ between platforms, compilers, builds and even runtimes.

Since we're not writing something like a driver, compilers get better and better, and we can get away with code that's a slightly less efficient (as long as the main bottleneck is somewhere else in the code), it's better to be safe than sorry

Reply:
Yes I agree, but we do initialize `i_n_n` and `base_i_` in the constructor of `AymLine`. So I'm wondering about a situation where this would occur? As AsymLine does not have a constructor that doesn't initialize these values. 

It could occur when of course in the future we would add another constructor.